### ISSUE TITLE
x second timer's image won't change when activated directly

### ISSUE THREAD

**Issue by OnsenManju** [NyxStudios/TShock#1374]
If 1/3/5 second timer was activated by right-clicking, its image won't turn the light on. It shows correctly on vanilla server or singleplayer.
**Vanilla server**
![](http://i.imgur.com/GLkkdSY.gif)
**TShock 4.3.22**
![](http://i.imgur.com/hWvkOXi.gif)
But when the timer was activated by other item (e.g. lever, switch, pressure plate), it lights on normally.
![](http://i.imgur.com/B6czhOD.gif)

**Comment by OnsenManju** [comment]
I just now noticed the lever also won't change its direction when being right-clicked.

**Comment by levi-middleton** [comment]
Yes. It seems like all the toggleables stay in their initial state.

**Comment by DogooFalchion** [comment]
Likely the same issue as campfires, these tiles need to be exempted from the right click->edit world protection tshock runs.  Makes me wonder if TSAPI used to insert a toggleable property like explosives is done.

**Comment by WhiteXZ** [comment]
Could not reproduce on bleeding edge build. Release however has this issue

**Comment by ProfessorXZ** [comment]
As mentioned in #1421, it's seems to be caused by the HitSwitch packet.
@OnsenManju would you mind testing it with this https://www.dropbox.com/s/duv3c5jyjpqqzib/TShockAPI.dll?dl=0

**Comment by OnsenManju** [comment]
@ProfessorXZ
Tested all [triggers](http://terraria.gamepedia.com/Mechanism) that have toggle-able images and can be activated by right-clicking.
![trigger_test](https://cloud.githubusercontent.com/assets/16373401/25181936/0fd13c22-254e-11e7-98c0-9bfb5affce80.gif)
```
              | TShock 4.3.23             | Test build      |
--------------+---------------------------+-----------------+
1/3/5 Timer   |                           |                 |
Switch        | image is unchanged        | same as vanilla |
Lever         |                           |                 |
--------------+---------------------------+-----------------+
Detonator     | won't revert once pressed | same as vanilla |
--------------+---------------------------+-----------------+
Trapped Chest | same as vanilla           | same as vanilla |
--------------+---------------------------+-----------------+
```
All of the above worked successfully with your test build. Awesome!
(Confirmed on Windows7 and Linux+Mono 4.8.0)

**Comment by ProfessorXZ** [comment]
@OnsenManju Thanks for the feedback, your reports are 10/10 :)
I'll PR the changes once the project is ready again.

**Comment by mistzzt** [comment]
This should fixed by latest commit #edad442e

**Comment by Ijwu** [comment]
Closing this on the basis that it is a very similar issue to #1421 and that issue was solved by edad442. Please re-open and comment if this is not fixed.
